libfreerdp/core/transport.c

```diff
@@ -380,7 +381,7 @@ BOOL transport_connect(rdpTransport* transport, const char* hostname,
 			}
 		}
 
-		if (!status && settings->GatewayRpcTransport)
+		if (!status && settings->GatewayRpcTransport && rpcFallback)
```

Comment:
Sure this is correct? Don't you want something like `if (!status && (settings->GatewayRpcTransport || rpcFallback))`?

Reply:
Nope, RPC is supposed to be used only if it's enabled via GatewayRpcTransport AND fallback to RPC makes sense (rpcFallback == TRUE).